### PR TITLE
Add: add math.pow

### PR DESCRIPTION
**Pull request by Jim8y**
Closes https://github.com/neo-project/neo-devpack-dotnet/issues/834

### CODE REVIEW

**Comment by Liaojinghui** [comment]
@lock9 please check

File: src/Neo.Compiler.CSharp/MethodConvert.cs [hunk 1]
```diff
@@ -4182,6 +4182,7 @@ private bool TryProcessSystemMethods(SemanticModel model, IMethodSymbol symbol,
                     AddInstruction(OpCode.SIGN);
                     return true;
                 case "System.Numerics.BigInteger.Pow(System.Numerics.BigInteger, int)":
+                case "System.Math.Pow(double, double)":
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert.cs:4185]: I think that we haven't got double or float compatibility

Comment (Liaojinghui) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4185]: yes and no. yes it uses double and neo does not support double. no it is c#, and it compiles and works well. truth is, we can not prevent developer from using double at all, it actually works in neo, c# convert them implecitly into int or other types, we cant prevent it, we cant detect it, we even 'support' it.

Comment (Liaojinghui) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4185]: i will do some study on this, maybe ban those key words from source level.

**Comment by Liaojinghui** [comment]
@shargon  now this can be implemented with https://github.com/neo-project/neo-devpack-dotnet/pull/839. the use of methods that take double or return double will be convertd to long by showing error highlight and suggestions.

**Comment by Liaojinghui** [comment]
Now we can get ride of `double`

File: src/Neo.SmartContract.Framework/Math.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+namespace Neo.SmartContract.Framework
+{
+    public class Math
+    {
+        [OpCode(OpCode.POW)]
+        public static extern BigInteger Pow(BigInteger x, BigInteger y);
+        [OpCode(OpCode.POW)]
+        public static extern BigInteger Pow(long x, long y);
+        [OpCode(OpCode.POW)]
+        public static extern BigInteger Pow(int x, int y);
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Math.cs:23]:
```suggestion
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(long x, long y);
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(int x, int y);
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(ulong x, ulong y);
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(uint x, uint y);
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(short x, short y);
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(ushort x, ushort y);
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(byte x, byte y);
```

File: src/Neo.SmartContract.Framework/Math.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        [OpCode(OpCode.POW)]
+        public static extern BigInteger Pow(ulong x, ulong y);
+        [OpCode(OpCode.POW)]
+        public static extern BigInteger Pow(uint x, uint y);
+        [OpCode(OpCode.POW)]
+        public static extern BigInteger Pow(short x, short y);
+        [OpCode(OpCode.POW)]
+        public static extern BigInteger Pow(ushort x, ushort y);
+        [OpCode(OpCode.POW)]
+        public static extern BigInteger Pow(byte x, byte y);
```
Comment (shargon) [inline src/Neo.SmartContract.Framework/Math.cs:33]:
```suggestion
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(byte x, byte y);
        [OpCode(OpCode.POW)]
        public static extern BigInteger Pow(sbyte x, sbyte y);
```

**Comment by Jim8y** [comment]
@dotnet-policy-service agree

File: src/Neo.Compiler.CSharp/MethodConvert.cs [hunk 4]
```diff
@@ -4513,44 +4513,16 @@ private bool TryProcessSystemMethods(SemanticModel model, IMethodSymbol symbol,
                         PrepareArgumentsForMethod(model, symbol, arguments);
                     Call(NativeContract.StdLib.Hash, "atoi", 1, true);
                     return true;
-                case "System.Math.Abs(sbyte)":
-                case "System.Math.Abs(short)":
-                case "System.Math.Abs(int)":
-                case "System.Math.Abs(long)":
```
Comment (shargon) [inline src/Neo.Compiler.CSharp/MethodConvert.cs:4519]: Why remove this compatibility?

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4519]: to avoid confusion, use `Neo.Math` everywhere instead of `System.Math`. Otherwise you will have two `Math` system.

Comment (cschuchardt88) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4519]: I prefer to use native `C#`. This may why developers can't program in `C#` _(I guess_). No internet search will tell them to use `Neo.math`, unless the `NC` error says to.

Comment (shargon) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4519]: Agree with @cschuchardt88

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4519]: You can't, many of them use `double`, i think you know what the issue is @shargon.
And I do not think using C# native is a good idea, this is neo C#, instead of dotnet C#, using dotnet C# causes confusing as people do not know what can be done and what can not.

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4519]: Where do you know what method is supported by neo @cschuchardt88?  only at compile time right? how does it feel? every time you use something, you know if it works only if you compile it......

Comment (cschuchardt88) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4519]: Understandable, Maybe we should phase out all native `dotnet` `APIs` and only have `Neo` `API` calls. For example, `Neo.Math.Abs`, `Neo.String` and etc

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4519]: Exactly what i have in mind.

File: src/Neo.Compiler.CSharp/MethodConvert.cs [hunk 5]
```diff
@@ -4513,44 +4513,16 @@ private bool TryProcessSystemMethods(SemanticModel model, IMethodSymbol symbol,
...
                     AddInstruction(OpCode.MAX);
                     return true;
-                case "System.Math.Min(byte, byte)":
-                case "System.Math.Min(sbyte, sbyte)":
-                case "System.Math.Min(short, short)":
-                case "System.Math.Min(ushort, ushort)":
-                case "System.Math.Min(int, int)":
-                case "System.Math.Min(uint, uint)":
-                case "System.Math.Min(long, long)":
-                case "System.Math.Min(ulong, ulong)":
```
Comment (vncoelho) [inline src/Neo.Compiler.CSharp/MethodConvert.cs:4553]: I think this part should be in another PR, @Jim8y .

It is a good point to discuss

Comment (Jim8y) [reply src/Neo.Compiler.CSharp/MethodConvert.cs:4553]: Sure, will move it to another pr.

**Comment by Jim8y** [comment]
now this pr only adds Math methods
